src/meta/client/MetaClient.cpp

```diff
@@ -1282,14 +1335,96 @@ MetaClient::getEdgeSchemaFromCache(GraphSpaceID spaceId, EdgeType edgeType, Sche
     }
 }
 
-StatusOr<const cpp2::IndexProperties>
-MetaClient::getTagIndexFromCache(TagIndexID) {
-    LOG(FATAL) << "Not implemented";
+StatusOr<std::shared_ptr<const cpp2::IndexFields>>
+MetaClient::getTagIndexFromCache(GraphSpaceID spaceId, TagIndexID tagIndexID) {
+    if (!ready_) {
+        return Status::Error("Not ready!");
+    }
+
+    folly::RWSpinLock::ReadHolder holder(localCacheLock_);
+    auto spaceIt = localCache_.find(spaceId);
+    if (spaceIt == localCache_.end()) {
+        // Not found
+        VLOG(3) << "Space " << spaceId << " not found!";
+        return Status::SpaceNotFound();
+    } else {
+        auto iter = spaceIt->second->tagIndexes_.find(tagIndexID);
+        if (iter == spaceIt->second->tagIndexes_.end()) {
+            VLOG(3) << "Space " << spaceId << ", Tag " << tagIndexID << " not found!";
+            return Status::TagNotFound();
+        } else {
+            return iter->second;
+        }
+    }
+}
```

Comment:
What is this based on tag?

Reply:
scan index via indexID , or write index data using indexId.

Reply:
IndexID ,tag|edgetype ID, list(ColumnDef) all are required for storage layer and graph layer.